### PR TITLE
Link some pages from footer

### PR DESCRIPTION
**Pull request by jonsgreen**
Link some pages from footer

# Reason For Change

From Miro Card: 
>As a designer I'd like some pages to be linked to in the footer of the site navigation, to take advantage of that real estate on a screen.

>I'd like to be able to group these pages into categories and have them displayed under their respective categories.

This PR attempts to satisfy the desire for grouping footer links to pages under categories in the footer with some minimal data modeling. For simplicity just a string array field was added for configuration on the website which populates a select list to choose from for a page. There is a slight cost of brittleness should category names need to be changed at some point but should hopefully be sufficient for MVP. 

This was also a good opportunity to more fully wire up the footer and header to better match the current design. Some improvements to image handling when missing and resizing for svgs was included as well. 

# Changes

- adds Website#footer_categories string array field populated with help
  of selectized js tags
- adds Page#footer_category to indicate that a page should be linked in
  a footer category
- wires up the footer category/links and other dynamic content and
  elements as well
- wires up the header navigation to more closely match the design
- fixes image resizing for svgs with resize_image_tag helper method
- ensures pages don't blow up with images are not attached

[Link some pages from footer](https://miro.com/app/board/uXjVO6C1LxA=/?moveToWidget=3458764523893158753&cot=14)

### CODE REVIEW

File: app/models/sponsor.rb [hunk 1]
```diff
@@ -25,3 +25,30 @@ def has_offer?
     offer_headline && offer_text && offer_url
   end
 end
+
+# == Schema Information
+#
+# Table name: sponsors
+#
```
Comment (codyjamesbrooks) [inline app/models/sponsor.rb:32]: What is the command to run this? I will try to remember to do it in the future.

File: app/views/layouts/themes/default/_header.html.haml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+%header#main-header{ **current_website.background_style }
```
Comment (codyjamesbrooks) [inline app/views/layouts/themes/default/_header.html.haml:1]: woah. What is that syntax?

File: app/assets/stylesheets/themes/default/footer.scss [hunk 3]
```diff
@@ -1,68 +1,132 @@
...
-
-  .footer-event-details {
-    border-bottom: 1px solid $divider-light;
-    padding-bottom: 20px;
+  color: white;
+  .overlay {
+    background-color: black !important;
+    opacity: 0.9 !important;
+    padding: 20px 30px 35px;
+  }
```
Comment (codyjamesbrooks) [inline app/assets/stylesheets/themes/default/footer.scss:7]: Were these colors not sticking without the '!important'?

File: app/decorators/website_decorator.rb [hunk 4]
```diff
@@ -32,4 +32,29 @@ def banner_sponsors
...
+
+  def register_page
+    pages.published.find_by(slug: 'register')
+  end
+
+  def background_style
+    return {} unless background.attached?
+
+    { style: "background-image: url('#{h.url_for(background)}');" }
+  end
```
Comment (codyjamesbrooks) [inline app/decorators/website_decorator.rb:59]: I now understand that cool syntax

File: app/assets/stylesheets/themes/default/footer.scss (hunk 3 above)
Comment (jonsgreen) [reply app/assets/stylesheets/themes/default/footer.scss:7]: Oh, I think I can remove those. Thanks for pointing that out.

**Review by codyjamesbrooks — COMMENTED** [review]
@jonsgreen I couldn't seem to get footer_categories to persist locally? Can we take a look at this together. I did got through everything and it all looks great.

**Pull request merged**